src/yarp.c

```diff
@@ -450,6 +450,8 @@ yp_parser_optional_constant_id_token(yp_parser_t *parser, const yp_token_t *toke
     return token->type == YP_TOKEN_NOT_PROVIDED ? 0 : yp_parser_constant_id_token(parser, token);
 }
 
+static size_t yp_statements_node_body_length(yp_statements_node_t *node);
```

Comment:
A small nit, but would you mind moving this prototype up with `yp_statements_node_create` and `yp_statements_node_body_append`?

Reply:
Done.  Since the grouping meant that the prototype is now declared a little later in the file, I've reverted one of the changes.